### PR TITLE
changefeedccl: warn when using https sink urls

### LINKED ISSUE #98719: changefeedccl: when using http sink, warn that it is not webhook
It is easy to use the http sink on accident, thinking it is the webhook sink, like what happened with a customer recently. Though we document this, most customers miss it (as https:// is what you would expect to use as the scheme, not webhook-https://). When creating a changefeed with the http sink, we should warn users and point to our webhook documentation.

Jira issue: CRDB-25462

Epic CRDB-28878

### PR DESCRIPTION
**Pull request by HonoreDB**
Changefeeds previously used http as their URI to align with backups.
Backups no longer support http, and for changefeeds http is ambiguous,
leading users to create cloudstorage sinks when they meant webhook sinks.
This PR changes the scheme to file-http. Webhooks still use webhook-http.

Fixes #98719.

Release note (enterprise change): Changefeeds that create files over an http connection may now be specified via `INTO 'file-https://'` to disambiguate with `webhook-https`.

### CODE REVIEW

File: pkg/ccl/changefeedccl/changefeed_stmt.go [hunk 1]
```diff
@@ -904,6 +904,15 @@ func validateSink(
 	if err != nil {
 		return err
 	}
+	u, err := url.Parse(details.SinkURI)
+	if err != nil {
+		return err
+	}
+
+	if u.Scheme == changefeedbase.SinkSchemeCloudStorageHTTP || u.Scheme == changefeedbase.SinkSchemeCloudStorageHTTPS {
```
Comment (miretskiy) [inline pkg/ccl/changefeedccl/changefeed_stmt.go:912]: Do we want to do that?
Are we going to bother people who want http with needles warning?

Would it be better to add file-http(s) scheme and prepare http(s) scheme for deprecation?

@amruss ?

File: pkg/ccl/changefeedccl/changefeed_stmt.go [hunk 2]
```diff
@@ -904,6 +904,18 @@ func validateSink(
...
 		return err
 	}
+	u, err := url.Parse(details.SinkURI)
+	if err != nil {
+		return err
+	}
+
+	if u.Scheme == changefeedbase.DeprecatedSinkSchemeHTTP || u.Scheme == changefeedbase.DeprecatedSinkSchemeHTTP {
+		p.BufferClientNotice(ctx, pgnotice.Newf(
+			`%s sinks will emit using cloud storage semantics. For a webhook sink, prepend webhook- to the sink URI. Prepend file- to remove this notice.`,
```
Comment (miretskiy) [inline pkg/ccl/changefeedccl/changefeed_stmt.go:914]: maybe rephrase a bit to avoid using "prepend"? Something like "Use webhook-https for webhook semantics, use file-http:// .."

**Review by miretskiy — APPROVED** [review]
Please update release note in commit and in PR to indicate deprecation.

File: pkg/ccl/changefeedccl/changefeed_stmt.go (hunk 2 above)
Comment (HonoreDB) [reply pkg/ccl/changefeedccl/changefeed_stmt.go]: Done.

**Comment by HonoreDB** [comment]
> Please update release note in commit and in PR to indicate deprecation.

Release note's in the second commit, merged it into the PR description.

**Comment by HonoreDB** [comment]
bors r=[miretskiy]
